### ISSUE TITLE
Zoom modal scratch window causes PaperWM, gnome-shell crash (libmutter involved?)

### ISSUE THREAD

**Issue by lediur** [paperwm/PaperWM#427]
Running into this issue after a Zoom call ends with the "Host has ended the meeting" modal. When the post-call Zoom modal opens, I get a crash that seems to cause PaperWM to crash and restart.

## Steps to reproduce

- Start a Zoom call on another device
- Join the Zoom call from the device running PaperWM
- Terminate the Zoom call from the other device

## Expected behavior

The "Host has ended the meeting" modal appears, and PaperWM does not crash

## Actual behavior

When the "Host has ended the meeting" modal appears, it has no window controls. PaperWM seems to crash and restart soon after.

## Error messages from `journalctl $(which gnome-shell)`

```
Oct 14 12:14:51 monolith gnome-shell[68048]: WARNING: window already registered Zoom
Oct 14 12:14:51 monolith gnome-shell[68048]: WARNING: window already registered Zoom
Oct 14 12:14:51 monolith gnome-shell[68048]: JS ERROR: undefined
                                              registerWindow@/home/derrick/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2273:15
                                             window_created@/home/derrick/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2214:14
Oct 14 12:14:51 monolith gnome-shell[68048]: JS ERROR: undefined
                                              registerWindow@/home/derrick/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2273:15
                                             window_created@/home/derrick/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2214:14
Oct 14 12:14:51 monolith gnome-shell[68048]: WARNING: window already registered Zoom
Oct 14 12:14:51 monolith gnome-shell[68048]: JS ERROR: undefined
                                              registerWindow@/home/derrick/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2273:15
                                             window_created@/home/derrick/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2214:14
Oct 14 12:14:51 monolith gnome-shell[68048]: WARNING: window already registered Zoom
Oct 14 12:14:51 monolith gnome-shell[68048]: JS ERROR: undefined
                                              registerWindow@/home/derrick/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2273:15
                                             window_created@/home/derrick/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2214:14
Oct 14 12:14:51 monolith gnome-shell[68048]: WARNING: window already registered Zoom
Oct 14 12:14:51 monolith gnome-shell[68048]: JS ERROR: undefined
                                              registerWindow@/home/derrick/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2273:15
                                             window_created@/home/derrick/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2214:14
Oct 14 12:14:51 monolith gnome-shell[68048]: WARNING: window already registered Zoom
Oct 14 12:14:51 monolith gnome-shell[68048]: JS ERROR: undefined
                                              registerWindow@/home/derrick/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2273:15
                                             window_created@/home/derrick/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2214:14
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <unnamed>[<MetaWindowGroup>:0x5629937345d0] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <unnamed>[<MetaBackgroundGroup>:0x562993734f60] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <spaceContainer>[<ClutterActor>:0x562998ec1900] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <clip>[<ClutterActor>:0x5629b6721150] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <space-actor>[<ClutterActor>:0x5629b6720e20] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <clone-clip>[<ClutterActor>:0x5629b6720af0] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <clone-container>[<StWidget>:0x562996315580] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <unnamed>[<ClutterActor>:0x562996829e50] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <unnamed>[<ClutterClone>:0x5629add78e80] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <unnamed>[<MetaWindowActorX11>:0x562998be1760] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <unnamed>[<MetaSurfaceActorX11>:0x562993b350d0] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <minimap-background>[<StWidget>:0x5629b389b1c0] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <container-clip>[<StWidget>:0x562996004230] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <minimap-container>[<StWidget>:0x5629bf6c5500] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <window-clone-container>[<ClutterActor>:0x5629990551b0] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: Can't update stage views actor <unnamed>[<ClutterClone>:0x5629a5e2bf80] is on because it needs an allocation.
Oct 14 12:14:51 monolith gnome-shell[68048]: **
Oct 14 12:14:51 monolith gnome-shell[68048]: libmutter:ERROR:../src/compositor/compositor.c:611:invalidate_top_window_actor_for_views: assertion failed: (!priv->frame_in_progress)
Oct 14 12:14:51 monolith gnome-shell[68048]: Bail out! libmutter:ERROR:../src/compositor/compositor.c:611:invalidate_top_window_actor_for_views: assertion failed: (!priv->frame_in_progress)
Oct 14 12:14:51 monolith gnome-shell[68048]: == Stack trace for context 0x562993044170 ==
Oct 14 12:14:51 monolith gnome-shell[68048]: #0   56299321cc90 i   resource:///org/gnome/shell/ui/main.js:737 (1aa8539bfa10 @ 243)
Oct 14 12:14:51 monolith gnome-shell[68048]: #1   56299321cb70 i   /home/derrick/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2539 (33826aae4c0 @ 670)
Oct 14 12:14:51 monolith gnome-shell[68048]: #2   56299321cad8 i   /home/derrick/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2249 (33826aae060 @ 45)
Oct 14 12:14:51 monolith gnome-shell[68048]: #3   56299321ca28 i   /home/derrick/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/utils.js:426 (33826ab2790 @ 160)
Oct 14 12:14:52 monolith gnome-shell[433484]: Running GNOME Shell (using mutter 43.0) as a X11 window and compositing manager
Oct 14 12:14:52 monolith gnome-shell[433484]: ATK Bridge is disabled but a11y has already been enabled.
Oct 14 12:14:52 monolith gnome-shell[433484]: JS WARNING: [resource:///org/gnome/gjs/modules/core/overrides/Gio.js 287]: Too many arguments to method Gio.AsyncInitable.init_async: expected 3, got 4
Oct 14 12:14:52 monolith gnome-shell[433484]: JS WARNING: [resource:///org/gnome/gjs/modules/core/overrides/Gio.js 287]: Too many arguments to method Gio.AsyncInitable.init_async: expected 3, got 4
Oct 14 12:14:52 monolith gnome-shell[433484]: Unset XDG_SESSION_ID, getCurrentSessionProxy() called outside a user session. Asking logind directly.
Oct 14 12:14:52 monolith gnome-shell[433484]: Will monitor session 2
```

## Environment

- `Linux monolith 5.19.0-2-amd64 #1 SMP PREEMPT_DYNAMIC Debian 5.19.11-1 (2022-09-24) x86_64 GNU/Linux
`
- Gnome Shell 43.0
- PaperWM at commit https://github.com/paperwm/PaperWM/commit/3bedb67f22881343c7b66639d28732fa6493d894

**Comment by ccope** [comment]
Zoom rapidly creates and destroys popup windows in my experience, not sure why exactly. It's probably triggering this race:

https://github.com/paperwm/PaperWM/blob/13ccdfa64d56da2e20f4adda2c0166109aa54397/tiling.js#L2240-L2250

As a hack, I wonder if adding a 50ms sleep here would help avoid the libmutter panic (that should give it time to render a few frames at least):
https://github.com/paperwm/PaperWM/blob/13ccdfa64d56da2e20f4adda2c0166109aa54397/tiling.js#L2274

**Comment by mjs** [comment]
I just hit the same crash (same libmutter assertion fail, same paperwm trace) except it occurred when a new Kitty window was being created. There was also a Bitwarden scratch window that had just been hidden but I don't know if that's related.

```
Nov 10 21:09:11 sibu gnome-shell[1168]: libmutter:ERROR:../mutter/src/compositor/compositor.c:611:invalidate_top_window_actor_for_views: assertion failed: (!priv->frame_in_progress)
Nov 10 21:09:11 sibu gnome-shell[1168]: Bail out! libmutter:ERROR:../mutter/src/compositor/compositor.c:611:invalidate_top_window_actor_for_views: assertion failed: (!priv->frame_in_progress)
Nov 10 21:09:11 sibu audit[1168]: ANOM_ABEND auid=1000 uid=1000 gid=1000 ses=4 pid=1168 comm="gnome-shell" exe="/usr/bin/gnome-shell" sig=6 res=1
Nov 10 21:09:11 sibu gnome-shell[1168]: == Stack trace for context 0x556a0d550320 ==
Nov 10 21:09:11 sibu gnome-shell[1168]: #0   556a0ee9c360 i   resource:///org/gnome/shell/ui/main.js:737 (8eb224bfa10 @ 243)
Nov 10 21:09:11 sibu gnome-shell[1168]: #1   556a0ee9c240 i   /usr/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2539 (2144e8954470 @ 670)
Nov 10 21:09:11 sibu gnome-shell[1168]: #2   556a0ee9c1a8 i   /usr/share/gnome-shell/extensions/paperwm@hedning:matrix.org/tiling.js:2249 (2144e8952fb0 @ 45)
Nov 10 21:09:11 sibu gnome-shell[1168]: #3   556a0ee9c0f8 i   /usr/share/gnome-shell/extensions/paperwm@hedning:matrix.org/utils.js:426 (2144e8957740 @ 160)
Nov 10 21:09:12 sibu systemd-coredump[2175]: Process 1168 (gnome-shell) of user 1000 dumped core.
                                             Module libgnome-shell-menu.so with build-id 70d2f659e3e33e3cc85bead389a8100d05de2eaf
                                             Module libgnome-shell.so with build-id 2230da3d01c8b7740767a7eeac4fcf914288ee55
                                             Module gnome-shell with build-id 8a0104e50990048192759eedf145edf0217022d9
                                             #2  0x0000556a0b3ffb19 n/a (gnome-shell + 0x2b19)
                                             #50 0x0000556a0b3ff48d n/a (gnome-shell + 0x248d)
                                             #53 0x0000556a0b3ff745 n/a (gnome-shell + 0x2745)
Nov 10 21:09:12 sibu gnome-shell[1725]: (EE) failed to read Wayland events: Broken pipe
```

**Comment by jwest75674** [comment]
Experiencing, I beleive, the same issue with Zoom, also with Google Meeting when trying to screen share. The small modal dialog loads fine, but if I interact with it, everything blinks for a bit before gnome crashes

**Comment by scottames** [comment]
> Experiencing, I beleive, the same issue with Zoom, also with Google Meeting when trying to screen share. The small modal dialog loads fine, but if I interact with it, everything blinks for a bit before gnome crashes

Same exact behavior here. Arch, Gnome 43, on the latest from the `develop` branch.

**Comment by hallettj** [comment]
I'm not sure, but maybe this would be fixed by #449. That PR fixes a crash that occurs when a floating window attempts to open a window.

**Comment by jtaala** [comment]
Yes, this is the precise error that #449 fixes.  I've also tested #449  with the reproduce steps mentioned here (with the zoom window tiled and in the scratch layer) and it does appear to fix this issue as well (i.e. no gnome crash etc. and things working as they should).

Could someone else confirm that PR #449 fixes this issue (after merging, running the `install.sh` script and logout/login etc.).
